addons/onedrive/client.py

```diff
@@ -27,22 +29,25 @@ def folders(self, folder_id=None):
         :return: a list of metadata objects representing the child folders of ``folder_id``
         """
 
-        if folder_id is None or folder_id == DEFAULT_ROOT_ID:
-            url = self._build_url(settings.ONEDRIVE_API_URL, 'drive', DEFAULT_ROOT_ID, 'children')
-        else:
-            url = self._build_url(settings.ONEDRIVE_API_URL, 'drive', 'items',
-                                  folder_id, 'children')
+        if drive_id is None:
+            raise Exception('drive_id is undefined, cannot proceed')
 
-        res = self._make_request(
+        folder_path_part = 'root' if folder_id is None else folder_id
+        list_folder_url = self._build_url(settings.MSGRAPH_API_URL, 'drives', drive_id,
+                                          'items', folder_path_part, 'children')
+        resp = self._make_request(
             'GET',
-            url,
+            list_folder_url,
+            headers=self._default_headers,
             params={'filter': 'folder ne null'},
             expects=(200, ),
             throws=HTTPError(401)
         )
-        return res.json()['value']
+        folder_list = resp.json()
+        logger.debug('folder_list:({})'.format(folder_list))
+        return folder_list['value']
```

Comment:
Is this extra debugging log intentional?

Reply:
Yep!   When debugging the provider, the most common thing I wanted to know was "what is OneDrive actually returning to us".   I've left these in, but downgraded the logging level to `debug` so they don't show up unless requested.